### PR TITLE
Fix warnings from compiling with Nvidia

### PR DESCRIPTION
**Pull request by psakievich**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [ ] Feature enhancement

## Description of the pull-request

Remove warnings when compiling with Nvidia in response to #588.  Also a little clean up and making it so Cuda space is now used when compiling for device.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [x] Linux
    - [ ] MacOS
  - Compilers 
    - [ ] GCC
    - [ ] LLVM/Clang
    - [ ] Intel compilers
    - [x] NVIDIA CUDA
- [x] Compiles without warnings
- [ ] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

File: include/actuator/ActuatorTypes.h [hunk 1]
```diff
@@ -16,9 +16,9 @@
 namespace sierra {
 namespace nalu {
 
-#ifdef ACTUATOR_ON_DEVICE
+#ifdef __CUDACC__
```
Comment (sayerhs) [inline include/actuator/ActuatorTypes.h:19]: We have been using `KOKKOS_ENABLE_CUDA` in other places, can you switch to that? Will be easier to replace it with whatever Kokkos suggests for `KOKKOS_ENABLE_<ACCELERATOR>`

Comment (psakievich) [reply include/actuator/ActuatorTypes.h:19]: Sure thing.

**Comment by psakievich** [comment]
@sayerhs also it looks like the `ActuatorSourceKernal` currently dies in the unittests so I don't think we're ready for a regression test yet.

File: include/actuator/ActuatorTypes.h [hunk 2]
```diff
@@ -16,15 +16,14 @@
 namespace sierra {
 namespace nalu {
 
-#ifdef ACTUATOR_ON_DEVICE
+#ifdef KOKKOS_ENABLED_CUDA
```
Comment (sayerhs) [inline include/actuator/ActuatorTypes.h:19]: I think there is a typo here. It is `KOKKOS_ENABLE_CUDA` not `ENABLED`

Comment (psakievich) [reply include/actuator/ActuatorTypes.h:19]: ah thanks

**Comment by psakievich** [comment]
@sayerhs right now it won't work. That's beyond the scope of what I was shooting for in this PR. With the last push I made all the non-fast actuator unit-tests are passing with a cuda build.  There should just be some small clean up to the kernels that are enabled when using fast to get it working though. I'd say let's merge this PR and open a new issue to address that goal.

**Pull request merged**